src/shared/regions/defaultRegionProvider.ts

```diff
@@ -29,14 +27,20 @@ export class DefaultRegionProvider implements RegionProvider {
         return this.onRegionProviderUpdatedEmitter.event
     }
 
-    public async getRegionData(): Promise<RegionInfo[]> {
-        return this._loadedRegions
-    }
-
     public isServiceInRegion(serviceId: string, regionId: string): boolean {
         return !!this.regionIdToRegionData.get(regionId)?.serviceIds.find(x => x === serviceId) ?? false
     }
 
+    public getPartitionId(regionId: string): string | undefined {
+        return this.regionIdToRegionData.get(regionId)?.partitionId ?? undefined
+    }
+
+    public getRegions(partitionId: string): Region[] {
+        return [...this.regionIdToRegionData.values()]
```

Comment:
Prly TS/JS noob question but, why do you have to put `values()`, which is already some sort of iterable, into another data structure before filtering/mapping? Does this perform a copy?

Reply:
It might be my limited understanding too... I tend to return arrays instead of iterators. Arrays have considerably more access methods.